### PR TITLE
another handful of fixes

### PR DESCRIPTION
**Pull request by slingamn**
The individual commits should be mostly self-documenting.

### CODE REVIEW

File: irc/server.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ type clientConn struct {
 func NewServer(config *Config, logger *logger.Manager) (*Server, error) {
 	// initialize data structures
 	server := &Server{
+		ctime:               time.Now().UTC(),
```
Comment (DanielOaks) [inline irc/server.go]: good find

**Comment by DanielOaks** [comment]
While I don't mind removing most of the indirection, I find it a little more interesting to explicitly need to declare the maps yourself now before safe use (unless you use the ApplyConfig, obv). Weird tradeoff of being indirect and having the maps defined for you vs direct and if you use 'em directly them you'll run into issues. Looks like it shouldn't be a drama though given how we use them (gosh I wish Go just make()'d the maps for you when you created a struct, that'd make it so dang easy).

**Comment by DanielOaks** [comment]
Great set of QoL fixes, thanks for the lovely PR

**Comment by slingamn** [comment]
Yeah, the pattern I've been going for in general is to replace `NewBazBat()` with `func (bb *BazBat) Initialize()`, which I have been thinking of as replacing one call with another (i.e., forgetting to call `NewBazBat()` is a nil pointer dereference, and forgetting to call `Initialize()` is some analogous panic), but the wider golang community and the stdlib seem to prefer the `NewBazBat` pattern and I'm not sure why.

I asked in #go-nuts about that map behavior and they called it [autovivification](https://en.wikipedia.org/wiki/Autovivification) and said it was bad, and I remember being moderately convinced at the time but I don't remember their arguments. https://github.com/goshuirc/irc-go/pull/11 did some autovivification-like behavior on the `IrcMessage` object (the tag maps became private members and were initialized only during the first relevant `SetTag` call).

**Pull request merged**